migration-guide.md

```diff
@@ -0,0 +1,317 @@
+- Context manager API for animation. Run `help(go.Figure().batch_animate)` for the full doc string.
+- Programmatic export of figures to static SVG images (and PNG and PDF with cairosvg installed).
+
+### Removed
+- Removed `.to_string`, `.strip_style`, `.get_data`, `.validate` and `.to_dataframe` methods from `plotly.graph_objs` objects. For example run `dir(plotly.graph_objs.Scatter)` to get all the (magic) methods of the Scatter class.
+
+
+### Changed
+- Improved data validation covering the full API with clear, informative error messages. This means that incorrect properties and/or values now always raise a `ValueError` with a description of the error, the invalid property, and the avaialble properties on the level that it was placed in the graph object. Eg. `go.Scatter(foo=123)` raises a validation error. See https://plot.ly/python/reference/ for a reference to all valid properties and values in the Python API.
+
+- graph objects are not `dict`s anymore. Running a cell of a graph object prints out a dict-style representation of the object:
+
+Eg. `plotly.graph_objs.Scatter()` prints
+
+```
+Scatter(**{
+    'type': 'scatter'
+})
+```
+
+- plotly objects now have a `.to_plotly_json` method that converts the object to a dict:
+
+Eg. `go.Scatter().to_plotly_json()` returns `{'type': 'scatter'}`
+
+
+
+### Deprecated
+- all graph objects must now be written using their full path. For example if one wants to customize the marker param in a scatter object, write `plotly.graph_objs.scatter.Marker` instead of `plotly.graph_objs.Marker`. If the marker object lives in a `plotly.graph_objs.Scatter()` object then a deprecated message will appear. Similarly
+
+```
+import plotly.graph_objs as go
+go.Scatter(
+    x=[0],
+    y=[0],
+    marker=go.Marker(
+        color='rgb(255,45,15)'
+    )
+)
+```
+
+produces a deprecation warning but
+
+```
+import plotly.graph_objs as go
+go.Scatter(
+    x=[0],
+    y=[0],
+    marker=go.scatter.Marker(
+        color='rgb(255,45,15)'
+    )
+)
+```
+
+does not.
+
+- `go.Data()` is deprecated. Use a list or array `[]` instead.
+
+
+# Getting Started
+## Installation
+To install and enable with Jupyter or Jupyter lab, run:
+```
+pip install plotly==3.0.0rc9
+pip install "notebook>=5.3" "ipywidgets>=7.2"  # only necessary for Jupyter Notebook environments
+```
+
+If you're using older versions of `notebook` or `ipywidgets` you may need to manually activate the widget extensions (this should not be needed for `notebook>=5.3` and `ipywidgets>=7.2`)
+
+```
+jupyter nbextension enable --py widgetsnbextension --sys-prefix
+jupyter nbextension enable --py plotlywidget --sys-prefix
+```
+
+In addition, to add JupyterLab support run the following commands
+
+```
+pip install jupyterlab
+jupyter labextension install @jupyter-widgets/jupyterlab-manager # install the Jupyter widgets extension
+jupyter labextension install plotlywidget
```

Comment:
Maybe add a note:
```
export NODE_OPTIONS=--max-old-space-size=4096
```
...to avoid issues being opened because of:
```
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```